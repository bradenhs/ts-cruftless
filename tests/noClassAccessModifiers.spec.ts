import { createSpec } from "./helpers/createSpec";

createSpec("noClassAccessModifiers");
