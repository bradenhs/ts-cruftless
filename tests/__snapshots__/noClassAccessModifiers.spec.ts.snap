// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`no-class-access-modifiers should emit no errors when source is valid 1`] = `""`;

exports[`no-class-access-modifiers should error when private is present 1`] = `
"/tests/cases/noClassAccessModifiers/shouldErrorWhenPrivateIsPresent.ts
ts-cruftless: The class access modifier \\"private\\" is not allowed. no-class-access-modifiers
  1 | class PrivateTest {
> 2 |   private foo = \\"bar\\";
    |  ^
  3 | }
  4 |"
`;

exports[`no-class-access-modifiers should error when protected is present 1`] = `
"/tests/cases/noClassAccessModifiers/shouldErrorWhenProtectedIsPresent.ts
ts-cruftless: The class access modifier \\"protected\\" is not allowed. no-class-access-modifiers
  1 | class ProtectedTest {
> 2 |   protected hello = 1;
    |  ^
  3 | }
  4 |"
`;

exports[`no-class-access-modifiers should error when public is present 1`] = `
"/tests/cases/noClassAccessModifiers/shouldErrorWhenPublicIsPresent.ts
ts-cruftless: The class access modifier \\"public\\" is not allowed. no-class-access-modifiers
  1 | class PublicTest {
> 2 |   public foo: string = \\"bar\\";
    |  ^
  3 | }
  4 |"
`;
